tensorflow_recommenders_addons/dynamic_embedding/python/ops/dynamic_embedding_optimizer.py

```diff
@@ -19,6 +19,7 @@
 from __future__ import print_function
 
 import functools
+import horovod.tensorflow as hvd
```

Comment:
It will introduce a new hvd dependency. Maybe use NcclAllReduce directly could avoid additional dependencies, isolated communicator problem, and performance cost of negotiation in horovod.

Reply:
I think it will be better to use horovod here. For CPU device, horovod will user `MPI_AllReduce`, for GPU device, horovod will use `NcclAllReduce`.

Reply:
Right. Some may do standalone or asynchronous training, without horovod. I think it would be better to only import horovod while using the synchronous training. Thus make the horovod  unbound.

Reply:
Hmm, I see. Thanks for your suggestion here.

Reply:
Maybe the following code can help.
```python
try:
  import horovod.tensorflow as hvd
except ImportError:
  pass

hvd.allreduce(...) # It will raise module not found error if hvd not installed.
```

Reply:
Thanks. Done now.